dss/collections/__init__.py

```diff
@@ -0,0 +1,109 @@
+import os
+
+from dss.config import Replica
+from dss.util.aws.clients import dynamodb  # type: ignore
+
+
+_collections_db_table_template = f"dss-collections-db-{{}}-{os.environ['DSS_DEPLOYMENT_STAGE']}"
+
+
+class CollectionData:
+    REPLICA = 'replica'
+    OWNER = 'owner'
+    UUID = 'uuid'
+    VERSION = 'version'
+
+
+def put_collection(doc: dict):
+    """Adds a new owner associated collection to the table if it does not already exist."""
+    dynamodb.put_item(
+        TableName=_collections_db_table_template.format(doc[CollectionData.REPLICA]),
+        Item={
+            'hash_key': {
+                'S': doc[CollectionData.OWNER]
+            },
+            'sort_key': {
+                'S': doc[CollectionData.UUID]
+            },
+            'versions': {
+                'S': doc[CollectionData.VERSION]  # comma-delimited string
```

Comment:
A string in DynamoDB is limited too 400KB: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Limits.html#limits-data-types

This creates an limit on the number of collections that can be associated with a single user in the DB. Consider a different scheme.

Reply:
Will modify.